### DISCUSSION TITLE
what is the origin story of helix?

### DISCUSSION THREAD

**Discussion by goyalyashpal** [helix-editor/helix#5789] [General]
many editors too have their origin story, so, i was wondering what inspired such a magnificent editor as helix?? and how does helix differ from kakoune? i mean i know helix's keybinds are even better, but what else?

it's kinda strange that why i am so drawn to origins of helix when there are many editors knowing whose origin story never crossed my mind lol...

- vim started as an improvement to vi
- nvim started as core refactoring to vim
- kakoune started as semantic refactoring to vim
- notepad3 started as cumulative modifications to ms-notepad
- vs code started as a refresh to visual studio and abusing new acquisition atom editor lol?
- texstudio started as a foss fork of some other software

**Comment by Omnikar** [comment]
I believe @archseer has said that Helix began as a personal project to satisfy his own editor needs, and ended up being picked up and built upon by others.

**Reply by kirawi** [reply]
I think part of it came from frustration over plugin configuration in Neovim.

**Reply by goyalyashpal** [reply]
- [v0.0.2] (2021-05-12) was the oldest helix release i could find here on gh, 
- ... wow, 2021... this project is _new_.
- I agree changing to year based "Calender Version Format `YY.0M(.MICRO)` in v[22.03] after v0.6 was a wise decision.

[v0.0.2]: https://github.com/helix-editor/helix/releases/tag/v0.0.2
[22.03]: https://github.com/helix-editor/helix/releases/tag/22.03

**Comment by BradKML** [comment]
How is Helix different from Micro and other CLI text editors? https://github.com/zyedidia/micro
Also why is every poweruser promoting Neovim or Emacs, and why would Helix break away from the crowd?

**Reply by IllyStable** [reply]
TLDR: Micro isn't a modal editor, while helix is. 

A modal editor (like vim, emacs, helix, - etc) uses different modes (e.g. insert, select, visual) and keybinds to navigate and alter text, normally making it alot more efficient keybind-wise - you don't need to maneouver the mouse to move to a different line, and instead (at least, in helix, just type `(line number)G`, and to delete a line you just type `xd` in normal mode. Generally modal editors take alot longer to learn but are worth the effort for the extra efficiency.
